### PR TITLE
perlPackages.ProcDaemon: init at 0.23

### PR DESCRIPTION
**Pull request by Jumail**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -20229,6 +20229,21 @@ with self; {
     };
   };
 
+  ProcDaemon = buildPerlPackage {
+    pname = "Proc-Daemon";
+    version = "0.23";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/A/AK/AKREAL/Proc-Daemon-0.23.tar.gz";
+      sha256 = "34c0b85b7948b431cbabc97cee580835e515ccf43badbd8339eb109474089b69";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:20237]:
```suggestion
      hash = "sha256-NMC4W3lItDHLq8l87lgINeUVzPQ7rb2DOesQlHQIm2k=";
```

**Review by stigtsp — CHANGES_REQUESTED** [review]
Hi there!

Some things:
- Please make sure the commit message follows [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md). For this addition it should be `perlPackages.ProcDaemon: init at 0.23`
- Also, src hashes in perl-packages.nix should be SRI hashes. You can convert hex hashes to SRI using `nix-hash`. I've converted the hex hash and added a code suggestion for that.

**Comment by Jumail** [comment]
Hi stigtsp!

Thanks for the response. I have updated everything according to the contributing guideline. 

According to the guideline it says ```meta.maintainers``` must be set, but since i'm not a maintainer of the project and other perl packages don't seem to have maintainers set, I'm not sure what to insert there. Let me know if that needs to be updated.

**Review by stigtsp — COMMENTED** [review]
Thanks!

Looks good, but could you please make the PR have only 1 commit with the message `perlPackages.ProcDaemon: init at 0.23` ?

**Comment by stigtsp** [comment]
> Thanks!
> 
> Looks good, but could you please make the PR have only 1 commit with the message `perlPackages.ProcDaemon: init at 0.23` ?

Hi there @Jumail ! Just revisiting this. If you can squash those to commits into a single commit with the same message, then I think we're good to go on this module.

**Comment by stigtsp** [comment]
Hi there! Thanks for the PR, perlPackages.ProcDaemon was meanwhile added by f6642aaf8b64bf6f74483938928f70452a3d79bc

Closing this

**Pull request closed without merging**